workflows/offline_ml_diags/offline_ml_diags/_plot_jacobian.py

```diff
@@ -11,22 +11,29 @@
 LINE_NAME = "jacobian_lines.png"
 
 
+def _separate_dimensions(jacobian_dict, dims):
+    pairs_2d, pairs_3d = [], []
+    for (input, output) in jacobian_dict.data_vars:
+        if jacobian_dict.sizes[input] == 1 or jacobian_dict.sizes[output] == 1:
+            pairs_2d.append((input, output))
+        elif jacobian_dict.sizes[input] > 1 and jacobian_dict.sizes[output] > 1:
+            pairs_3d.append((input, output))
+    return pairs_2d, pairs_3d
+
+
 def plot_jacobian(model: fv3fit.keras._models.DenseModel, output_dir: str):
     jacobian_dict = model.jacobian()
 
-    data_vars: Tuple[str, str] = jacobian_dict.data_vars  # type: ignore
+    pairs_2d, pairs_3d = _separate_dimensions(jacobian_dict, 2)
+    inputs_2d, outputs_2d = {in_name for in_name, out_name in pairs_2d}, {out_name for in_name, out_name in pairs_2d}
+    inputs_3d, outputs_3d = {in_name for in_name, out_name in pairs_3d}, {out_name for in_name, out_name in pairs_3d}
```

Comment:
Reverted to the original since the more concise version needed if statement to handle zero length case, which then required a fix for the type checker.